category/category.go

```diff
@@ -0,0 +1,269 @@
+		}
+		log.Error(ctx, map[string]interface{}{
+			"category_id": category.ID,
+			"err":         db.Error.Error(),
+		}, "unable to create category")
+		return nil, errors.NewInternalError(db.Error)
+	}
+	log.Info(ctx, map[string]interface{}{
+		"category_id": category.ID,
+	}, "category created successfully")
+	return category, nil
+}
+
+// LoadAllRelationshipsOfCategory loads all the relationships of a category. This is required for workitemtype filtering.
+func (m *GormRepository) LoadAllRelationshipsOfCategory(ctx context.Context, categoryID uuid.UUID) ([]*WorkItemTypeCategoryRelationship, error) {
+	// Check if category is present
+	_, err := m.LoadCategoryFromDB(ctx, categoryID)
+	if err != nil {
+		return nil, errs.Wrap(err, fmt.Sprintf("failed to load category with id %s", categoryID))
+	}
+
+	relationship := []*WorkItemTypeCategoryRelationship{}
+	db := m.db.Model(&relationship).Where("category_id=?", categoryID).Find(&relationship)
+	if db.RecordNotFound() {
+		log.Error(ctx, map[string]interface{}{
+			"category_id": categoryID,
+		}, "workitemtypes of category not found")
+		return nil, errors.NewNotFoundError("work item type category", categoryID.String())
+	}
+	if err := db.Error; err != nil {
+		log.Error(ctx, map[string]interface{}{
+			"category_id": categoryID,
+			"err":         err,
+		}, "unable to list workitemtype category relationships")
+		return nil, errors.NewInternalError(db.Error)
+	}
+	return relationship, nil
+}
+
+// LoadCategoryFromDB returns category for the given id
+// This is needed to check if a category is present in db or not.
+func (m *GormRepository) LoadCategoryFromDB(ctx context.Context, id uuid.UUID) (*Category, error) {
+	res := Category{}
+	db := m.db.Model(&res).Where("id=?", id).First(&res)
+	if db.RecordNotFound() {
+		log.Error(ctx, map[string]interface{}{
+			"category_id": id,
+		}, "category not found")
+		return nil, errors.NewNotFoundError("category", id.String())
+	}
+	if err := db.Error; err != nil {
+		log.Error(ctx, map[string]interface{}{
+			"category_id": id,
+			"err":         err,
+		}, "unable to load category", err.Error())
+		return nil, errors.NewInternalError(err)
+	}
+	return &res, nil
+}
+
+// LoadWorkItemTypeCategoryRelationship loads all the relationships of a category. This is required for testing.
+func (m *GormRepository) LoadWorkItemTypeCategoryRelationship(ctx context.Context, workitemtypeID uuid.UUID, categoryID uuid.UUID) (*WorkItemTypeCategoryRelationship, error) {
+	// Check if category is present
+	_, err := m.LoadCategoryFromDB(ctx, categoryID)
+	if err != nil {
+		log.Error(ctx, map[string]interface{}{
+			"category_id": categoryID,
+		}, "category not found")
+		return nil, errs.Wrap(err, fmt.Sprintf("failed to load category with id %s", categoryID))
+	}
+	relationship := WorkItemTypeCategoryRelationship{}
+	db := m.db.Model(&relationship).Where("category_id=? AND work_item_type_id=?", categoryID, workitemtypeID).Find(&relationship)
```

Comment:
I will explain here:

1. I have a category ID

2. I want to create a WIT and associate it with a category. So, while creating WIT, I will pass the category ID -> https://github.com/almighty/almighty-core/pull/1037/files/#diff-cdc5917bec757bac125fd9bf5c0ef0c1R167

3. This (a) creates the WIT (b) creates a relationship between a WIT and a category (i.e. it adds a row in work_item_type_categories table with the WIT id and category id)

4. Now when a WIT is created, it returns a **WorkItemType** object (https://github.com/almighty/almighty-core/pull/1037/files/#diff-cdc5917bec757bac125fd9bf5c0ef0c1R162)  
It doesnot return a **WorkItemTypeCategoryRelationship** object (https://github.com/almighty/almighty-core/pull/1037/files/#diff-194905cabff622d515b0957a60cba773R49)

So, the reason that why i use both "categoryID" and "WIT id" to load a relationship is because:
1. I can't simply use category_id because a category can have many WITs associated with it. While testing, I just need that specific relationship.
2. I can't simply use workitemtype_id because a WIT can belong to many categories.